🌐 Complete missing Hindi translations for Ghost i18n

dawgdevv opened the pull request:
This PR completes all missing Hindi translations across Ghost's i18n files, ensuring 100% translation coverage for the Hindi locale.

### Changes:
- ✅ Completed missing translations in `portal.json` (58 strings)
- ✅ Completed missing translations in `search.json` (7 string)
- ✅ Verified all other Hindi locale files are already complete
- ✅ All tests pass and translations are validated

### Files Updated:
- `ghost/i18n/locales/hi/portal.json`
- `ghost/i18n/locales/hi/search.json`
- `ghost/i18n/locales/context.json` (auto-updated by test suite)

ref #23361

dawgdevv committed:
🌐 Complete Hindi translations for Ghost i18n

- Complete missing translations in portal.json and search.json
- All Hindi locale files now 100% translated (379 total strings)
- Maintain consistency with existing Hindi translations

This completes the Hindi language support for Ghost's user interface,
providing full localization for Hindi-speaking users.

dawgdevv committed:
🌐 Update i18n context file after running tests

Auto-generated update to context.json from i18n test suite

dawgdevv committed:
Merge branch 'main' into feature/i18n/hindi-missing-translations

dawgdevv commented:
Hello @cathysarisky  can you review this i have completed all the hindi translation

cathysarisky commented on ghost/i18n/locales/hi/portal.json:
```diff
@@ -10,16 +10,16 @@
-    "An error occurred": "",
-    "An unexpected error occured. Please try again or <a>contact support</a> if the error persists.": "एक अप्रत्याशित त्रुटि हुई। कृपया फिर से प्रयास करें या यदि त्रुटि बनी रहती है तो <a>समर्थन से संपर्क करें</a>।",
+    "An error occurred": "एक गड़बड़ी हुई",
+    "An unexpected error occured. Please try again or <a>contact support</a> if the error persists.": "एक अप्रत्याशित गड़बड़ी हुई। कृपया फिर से प्रयास करें या यदि गड़बड़ी बनी रहती है तो <a>समर्थन से संपर्क करें</a>।",
```
🤖 AI: I suspect using “त्रुटि” instead of “गड़बड़ी” might align better with technical UI language. Could you consider changing “एक अप्रत्याशित गड़बड़ी हुई।” to something like “एक अप्रत्याशित त्रुटि हुई।”?

cathysarisky commented on ghost/i18n/locales/hi/portal.json:
```diff
@@ -58,28 +58,28 @@
+    "Failed to process checkout, please try again": "चेकआउट करने में विफल, कृपया पुनः प्रयास करें",
+    "Failed to send magic link email": "जादुई लिंक ईमेल भेजने में विफल",
+    "Failed to send verification email": "सत्यापन ईमेल भेजने में विफल",
+    "Failed to sign up, please try again": "साइन अप करने में विफल, कृपया पुनः प्रयास करें",
```
🤖 AI: I noticed you used “साइन अप” here, but elsewhere you have “साइनअप” and “साइन-अप”. Could you pick one form and use it consistently across portal translations?

cathysarisky commented on ghost/i18n/locales/hi/portal.json:
```diff
@@ -139,10 +139,10 @@
-    "Signups from this email domain are currently restricted.": "",
-    "Something went wrong, please try again later.": "",
-    "Sorry, no recommendations are available right now.": "",
+    "Signup error: Invalid link": "साइनअप गड़बड़ी: अमान्य लिंक",
```
🤖 AI: Here you use “साइनअप” without space for “Signup error”. To keep consistency, should this match your choice at diffPosition 90 (“साइन अप”) or use a hyphenated form?

cathysarisky commented on ghost/i18n/locales/hi/portal.json:
```diff
@@ -139,10 +139,10 @@
-    "Something went wrong, please try again later.": "",
-    "Sorry, no recommendations are available right now.": "",
+    "Signup error: Invalid link": "साइनअप गड़बड़ी: अमान्य लिंक",
+    "Signups from this email domain are currently restricted.": "इस ईमेल डोमेन से साइनअप वर्तमान में प्रतिबंधित हैं।",
```
🤖 AI: This line uses “साइनअप” for “Signups from this email domain”. Please ensure this matches the chosen style for “sign up” across all strings.

cathysarisky commented on ghost/i18n/locales/hi/portal.json:
```diff
@@ -97,33 +97,33 @@
-    "Jamie Larson": "",
-    "jamie@example.com": "",
+    "Invalid email address": "अमान्य ईमेल पता",
+    "Jamie Larson": "जेमी लार्सन",
```
🤖 AI: According to guidelines, we should not transliterate placeholder names like “Jamie Larson”. Could you replace “जेमी लार्सन” with a common unisex name in Hindi?

cathysarisky commented on ghost/i18n/locales/hi/portal.json:
```diff
@@ -151,38 +151,38 @@
-    "This site is not accepting donations at the moment.": "",
-    "This site is not accepting payments at the moment.": "",
-    "This site only accepts paid members.": "",
+    "This site is not accepting donations at the moment.": "यह साइट इस समय दान स्वीकार नहीं कर रही है",
```
🤖 AI: The English string ends with a period. Please add a final punctuation mark (Hindi danda “।”) at the end of the translation.

cathysarisky commented on ghost/i18n/locales/hi/portal.json:
```diff
@@ -151,38 +151,38 @@
-    "This site is not accepting payments at the moment.": "",
-    "This site only accepts paid members.": "",
+    "This site is not accepting donations at the moment.": "यह साइट इस समय दान स्वीकार नहीं कर रही है",
+    "This site is not accepting payments at the moment.": "यह साइट इस समय भुगतान स्वीकार नहीं कर रही है",
```
🤖 AI: The English string ends with a period. Please add a final punctuation mark (Hindi danda “।”) at the end of the translation.

cathysarisky commented on ghost/i18n/locales/hi/portal.json:
```diff
@@ -151,38 +151,38 @@
-    "This site only accepts paid members.": "",
+    "This site is not accepting donations at the moment.": "यह साइट इस समय दान स्वीकार नहीं कर रही है",
+    "This site is not accepting payments at the moment.": "यह साइट इस समय भुगतान स्वीकार नहीं कर रही है",
+    "This site only accepts paid members.": "यह साइट केवल भुगतान करने वाले सदस्यों को स्वीकार करती है",
```
🤖 AI: The English string ends with a period. Please add a final punctuation mark (Hindi danda “।”) at the end of the translation.

cathysarisky commented on ghost/i18n/locales/hi/portal.json:
```diff
@@ -151,38 +151,38 @@
-    "Too many different sign-in attempts, try again in {number} hours": "",
-    "Too many different sign-in attempts, try again in {number} minutes": "",
-    "Too many sign-up attempts, try again later": "",
+    "To continue to stay up to date, subscribe to {publication} below.": "अप-टू-डेट रहने के लिए, नीचे दिए गए {publication} की सदस्यता लें",
```
🤖 AI: The English string ends with a period. Please add a final punctuation mark (Hindi danda “।”) at the end of the translation.

cathysarisky commented on ghost/i18n/locales/hi/search.json:
```diff
@@ -1,9 +1,9 @@
+    "Cancel": "रद्द करें",
+    "No matches found": "कोई मैच नहीं मिला",
+    "Posts": "पोस्ट",
+    "Search posts, tags and authors": "लेख, टैग और लेखकों की तलाश करें",
```
🤖 AI: For consistency with other translations where “Posts” is translated as “पोस्ट”, would it be better to use “पोस्ट, टैग और लेखक” instead of “लेख, टैग और लेखकों”?

cathysarisky reviewed (changes requested):
📋 My AI helper 🤖 left you a few comments.  I always believe the human over the AI, so feel free to mark as resolved anything that isn't on target.  I find the AI is good at marking areas that might need to be checked again, but you don't have to make the suggested changes if they don't make sense.  You speak better Hindi than it does! 

I'm hoping that @rohit-bhardwajj will have a look also!  Thanks for collaborating!

 Please leave me a comment when you're satisfied with everything. :) 

🤖 AI: Good job overall! I noticed a few consistency items—especially around punctuation (adding Hindi danda where the English has a period), uniform treatment of “sign up” (साइन अप vs साइनअप vs साइन-अप), and placeholder standards for names and digits. Once these are aligned, the translations should feel very smooth. धन्यवाद!

---
*Drafted with my new i18n tooling - feedback welcome*

rohit-bhardwajj reviewed (commented):
Hey! Great job on completing the translations — it's looking solid overall.

Just had few suggestions:

1. For this line:
   > `"To continue to stay up to date, subscribe to {publication} below.": "अप-टू-डेट रहने के लिए, नीचे दिए गए {publication} की सदस्यता लें"`

   Consider replacing **"अप-टू-डेट"** with something more native and formal like **"नवीनतम जानकारी प्राप्त करने के लिए"** — this aligns better with the rest of the UI tone.

2. Also noticed the phone number:
   > `"+1 (123) 456-7890": "+१ (१२३) ४५६-७८९०"`

   It might be better to **retain standard English numerals** in this context, especially since phone numbers are often expected in international format. Translating digits may reduce clarity or usability.

cathysarisky commented:
>    It might be better to **retain standard English numerals** in this context, especially since phone numbers are often expected in international format. Translating digits may reduce clarity or usability.

The phone number field is not in use anywhere. So let's just leave it untranslated.

dawgdevv committed:
Merge branch 'main' into feature/i18n/hindi-missing-translations

dawgdevv committed:
🌐 Update Hindi translations for signup and error messages

### Changes:
- All the changes that are requested are added.
- All the request to promote more native feel to UI are addressed
- Removed the phone number translated into native language  because it not used.
